pkg/expression/context/context.go

```diff
@@ -234,3 +243,22 @@ func AssertLocationWithSessionVars(ctxLoc *time.Location, vars *variable.Session
 		ctxLocStr, varsLocStr, stmtLocStr,
 	)
 }
+
+// StaticalizableExprContext provides more methods to implement the clone of a `ExprContext`
+type StaticalizableExprContext interface {
+	ExprContext
+
+	GetStaticalizableEvalContext() StaticalizableEvalContext
```

Comment:
Could we only give one method like `GetSpecailStaticalizableExprCtxOptions() ([]StaticExprCtxOption, []StaticEvalCtxOption)` to return some special fields to staticalize.

I think its can reduce same misusages like `GetPlanCacheTracker` which should only be used in `Detach`